src/main/scala/rocket/ScratchpadSlavePort.scala

```diff
@@ -97,22 +96,25 @@ class ScratchpadSlavePort(address: AddressSet)(implicit p: Parameters) extends L
 }
 
 /** Mix-ins for constructing tiles that have optional scratchpads */
-trait CanHaveScratchpad extends HasHellaCache with HasICacheFrontend with HasCoreParameters {
+trait CanHaveScratchpad extends HasHellaCache with HasICacheFrontend {
   val module: CanHaveScratchpadModule
+  val xLenBytes = p(XLen)/8
+  val cacheBlockBytes = p(CacheBlockBytes)
+  val fetchBytes = tileParams.core.fetchBytes
 
   val slaveNode = TLInputNode() // Up to two uses for this input node:
 
   // 1) Frontend always exists, but may or may not have a scratchpad node
-  val fg = LazyModule(new TLFragmenter(fetchWidth*coreInstBytes, p(CacheBlockBytes), earlyAck=true))
-  val ww = LazyModule(new TLWidthWidget(xLen/8))
+  val fg = LazyModule(new TLFragmenter(fetchBytes, cacheBlockBytes, earlyAck=true)) // TODO: should these be moved outside the tile &
+  val ww = LazyModule(new TLWidthWidget(xLenBytes))                                 //       generated by a new PeripheryBus method?
```

Comment:
If the tile remains a boundary for hierarchical physical design, then these should stay within the tile, to reduce the number of wires crossing that boundary.

Reply:
Depends on whether you expect the PeripheryBus to be wider or narrower than `xBytes`... Also as we discussed I think these should actually both go on some kind of intra-tile front bus

Reply:
Yeah my misunderstanding is that there wasn't already an intra-tile bus.  That seems to me to be the right decision (in a separate PR).